power_grid_model_c/power_grid_model/include/power_grid_model/component/fault.hpp

```diff
@@ -125,6 +125,22 @@ class Fault final : public Base {
         return fault_type_;
     }
     FaultPhase get_fault_phase() const {
```

Comment:
This function should be called in `calc_param`, there you assign `fault_type` and `fault_phase` to the math input.